crates/biome_cli/src/execute/process_file/workspace_file.rs

```diff
@@ -60,9 +62,22 @@ impl<'ctx, 'app> WorkspaceFile<'ctx, 'app> {
         self.guard().get_file_content()
     }
 
+    pub(crate) fn as_extension(&self) -> Option<&str> {
+        self.path.extension().and_then(|s| s.to_str())
+    }
+
     /// It updates the workspace file with `new_content`
     pub(crate) fn update_file(&mut self, new_content: impl Into<String>) -> Result<(), Error> {
-        let new_content = new_content.into();
+        let mut new_content = new_content.into();
+        if self.as_extension() == Some("astro") {
+            let mut edges = ASTRO_FENCE.find_iter(&self.input);
+            if let (Some(start), Some(end)) = (edges.next(), edges.next()) {
+                let mut tmp = self.input.clone();
+                tmp.replace_range(start.end()..end.start(), new_content.as_str());
+                new_content = tmp;
```

Comment:
I am not very fond of this workaround, where we need a temporary variable. Is there a better way to do so?

Reply:
I haven’t tested this, but couldn’t you do something like:

```rs
new_content = format!(“{}{}{}”,
    self.input[..start.end()],
    new_content,
    self.input[end.start()..]
);
```

Reply:
Effective! I'll do that :)